src/main/java/de/tum/in/www1/artemis/service/plagiarism/TextPlagiarismResultConverter.java

```diff
@@ -0,0 +1,106 @@
+        textPlagiarismResult.setExercise(exercise);
+
+        return textPlagiarismResult;
+    }
+
+    private static void verifyAllowedExerciseType(Exercise exercise) {
+        if (!(exercise instanceof TextExercise || exercise instanceof ProgrammingExercise)) {
+            throw new IllegalStateException("TextPlagiarismResults are supported only for TextExercise and ProgrammingExercise.");
+        }
+    }
+
+    /**
+     * Create a new PlagiarismComparison instance from an existing JPlagComparison object.
+     *
+     * @param jplagComparison     JPlag comparison to map to the new PlagiarismComparison instance
+     * @param exercise            the exercise to which the comparison belongs, either Text or Programming
+     * @param submissionDirectory the directory to which all student submissions have been downloaded / stored
+     * @return a new instance with the content of the JPlagComparison
+     */
+    private PlagiarismComparison<TextSubmissionElement> fromJPlagComparison(JPlagComparison jplagComparison, Exercise exercise, File submissionDirectory) {
+        PlagiarismComparison<TextSubmissionElement> comparison = new PlagiarismComparison<>();
+
+        comparison.setSubmissionA(createSubmissionFromJPlagSubmission(jplagComparison.firstSubmission(), exercise, submissionDirectory));
+        comparison.setSubmissionB(createSubmissionFromJPlagSubmission(jplagComparison.secondSubmission(), exercise, submissionDirectory));
+        comparison.setMatches(jplagComparison.matches().stream().map(PlagiarismMatch::fromJPlagMatch).collect(Collectors.toSet()));
+        // Note: JPlag returns a value between 0 and 1, we assume and store a value between 0 and 100 (percentage) in the database
+        comparison.setSimilarity(jplagComparison.similarity() * 100);
+        comparison.setStatus(PlagiarismStatus.NONE);
+
+        return comparison;
+    }
+
+    /**
+     * Create a new PlagiarismSubmission instance from an existing JPlag Submission
+     *
+     * @param jplagSubmission     the JPlag Submission to create the PlagiarismSubmission from
+     * @param exercise            the exercise to which the comparison belongs, either Text or Programming
+     * @param submissionDirectory the directory to which all student submissions have been downloaded / stored
+     * @return a new PlagiarismSubmission instance
+     */
+    private PlagiarismSubmission<TextSubmissionElement> createSubmissionFromJPlagSubmission(Submission jplagSubmission, Exercise exercise, File submissionDirectory) {
+        var submission = new PlagiarismSubmission<TextSubmissionElement>();
+
+        submission.setElements(jplagSubmission.getTokenList().stream().filter(Objects::nonNull)
+                .map(token -> TextSubmissionElement.fromJPlagToken(token, submission, exercise, submissionDirectory)).collect(Collectors.toCollection(ArrayList::new)));
+        submission.setSize(jplagSubmission.getNumberOfTokens());
+        submission.setScore(null); // TODO
```

Comment:
TODO?

Reply:
This was there before. This logic is copied from model classes

Reply:
I removed it, as probably no one know what it meant